controllers/openstackmachine_controller.go

```diff
@@ -463,7 +466,7 @@ func machineToInstanceSpec(openStackCluster *infrav1.OpenStackCluster, machine *
 		ConfigDrive:            openStackMachine.Spec.ConfigDrive != nil && *openStackMachine.Spec.ConfigDrive,
 		RootVolume:             openStackMachine.Spec.RootVolume,
 		AdditionalBlockDevices: openStackMachine.Spec.AdditionalBlockDevices,
-		ServerGroupID:          openStackMachine.Spec.ServerGroupID,
+		ServerGroupID:          openStackMachine.Status.ReferencedResources.ServerGroupID,
```

Comment:
This isn't set anywhere, if I read the PR correctly. Status of a machine is usually set after a machine is created. I'll take a deeper look.